### PR TITLE
implement MicroProfile Fault Tolerance 2.1

### LINKED ISSUE #146: implement MicroProfile Fault Tolerance 2.1 changes
That is:

- added `CircuitBreaker.skipOn`
- added `Fallback.applyOn` and `Falback.skipOn`
- `Retry.retryOn` and `Retry.abortOn` no longer ignore `Throwable.class` (should be enough to just remove `SetOfThrowables.withoutCustomThrowables` and replace its usages with `create`)
- behavior of overlapping `Retry.retryOn` and `Retry.abortOn` was specified (we should be fine, and if not, the TCK will catch it)

### PR DESCRIPTION
**Pull request by Ladicek**
Resolves #146.

### CODE REVIEW

File: implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/FaultToleranceInterceptor.java [hunk 1]
```diff
@@ -158,6 +164,10 @@ public Object interceptCommand(InvocationContext invocationContext) throws Excep
                 CompletableFuture<T> result = new CompletableFuture<>();
                 asyncExecutor.submit(() -> {
                     try {
+                        // the requestContextController.activate/deactivate pair here is the minimum
+                        // to pass TCK; for anything serious, Context Propagation is required!
+                        // TODO does this code work together with Context Propagation?
+                        requestContextController.activate();
```
Comment (Ladicek) [inline implementation/fault-tolerance/src/main/java/io/smallrye/faulttolerance/FaultToleranceInterceptor.java]: I'd be mostly interested in your thoughts about this. I'm pretty sure the `Future`  case is fine, but here, I'm not sure.

File: pom.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
     <version.hdrHistogram>2.1.9</version.hdrHistogram>
     <version.jakarta.enterprise.concurrent-api>1.1.2</version.jakarta.enterprise.concurrent-api>
     <version.jakarta.interceptor-api>1.2.5</version.jakarta.interceptor-api>
-    <version.mp-fault-tolerance>2.0.3</version.mp-fault-tolerance>
+    <version.mp-fault-tolerance>2.1-RC2</version.mp-fault-tolerance>
```
Comment (rhusar) [inline pom.xml]: 2.1 final is now out: https://github.com/eclipse/microprofile-fault-tolerance/releases/tag/2.1

Comment (Ladicek) [reply pom.xml]: Right. I'm just testing one solution for the request context activation in presence/absence of Context Propagation, if it works, this PR will be ready for merging today :-)

Comment (rhusar) [reply pom.xml]: Great, just don't forget to branch before that :-) if that's still the plan.

Comment (kenfinnigan) [reply pom.xml]: Why branch before release?

Comment (rhusar) [reply pom.xml]: From what I understood, SR FT 4.0.x is already used in Quarkus following FT 2.0 spec. Thus a SR FT 4.1.x would be following the FT 2.1 spec.

Comment (kenfinnigan) [reply pom.xml]: Ok, but why is a branch needed?

I don't believe anything other than Quarkus is using 4.0.x, so we don't need to retain it for anything

Comment (Ladicek) [reply pom.xml]: Yea, we don't need a branch for 4.0 at this point in time. We can always create one retroactively, if there's a need.

Comment (rhusar) [reply pom.xml]: In general what we do in MW is branches are created to support different versions used in different projects. So in this instance, in order to support that Quarkus version using the SR FT 4.0.0 a "4.0.x" branch is created. If you think there will never be a 4.0.1 then certainly branch is not necessary.

Comment (kenfinnigan) [reply pom.xml]: For SmallRye we've taken the opposite approach, to only create branches when needed.

The reason is that most often a new release comes out and is used in downstream projects before any updates to an older release would be needed.

In this particular case, as Quarkus will update to the new 4.1.x release, we don't need a 4.0.x branch

**Comment by Ladicek** [comment]
This PR should be final now. It uses MP FT 2.1 final release and passes TCK both with and without Context Propagation. Yay :-)

File: pom.xml (hunk 2 above)
Comment (rhusar) [reply pom.xml]: Ok, although it's confusing and might fly for projects like Quarkus, but not branches that will be used in EAP given it is supported for a long time.

Comment (kenfinnigan) [reply pom.xml]: Any SmallRye implementation used in EAP, won't be supported for the same length as regular EAP.

It's something we still need to figure out

**Pull request merged**